Wikimate.php

```diff
@@ -858,3 +858,501 @@ private function findSection($section)
 		return -1;
 	}
 }
+
+
+/**
+ * Models a wiki file that can have its properties retrieved and
+ * its contents downloaded and uploaded.
+ * All properties pertain to the current revision of the file.
+ *
+ * @author  Robert McLeod & Frans P. de Vries
+ * @since   October 2016
+ */
+class WikiFile
+{
+	protected $filename  = null;
+	protected $wikimate  = null;
+	protected $exists    = false;
+	protected $invalid   = false;
+	protected $error     = null;
+	protected $edittoken = null;
+	protected $info      = null;
+
+	/*
+	 *
+	 * Magic methods
+	 *
+	 */
+
+	/**
+	 * Constructs a WikiFile object from the filename given
+	 * and adds a Wikimate object.
```

Comment:
Just like in the WikiPage constructor, this is related to how Wikimate and the other two classes are strung together. `Wikimate::getFile` passes `$this` (PHP's variable for the current object, i.e. Wikimate) to the WikiFile constructor, so that WikiFile methods can then access the underlying Wikimate methods.

The user isn't supposed to called the WikiPage/WikiFile constructors directly, so I'm not sure how to clarify this, nor whether it's needed.